packages/components/src/navigation/types.ts

```diff
@@ -0,0 +1,316 @@
+	/**
+	 * When `true`, enables the search feature on the menu title.
+	 */
+	hasSearch?: boolean;
+	/**
+	 * Indicates whether the menu is empty or not. Used together with the
+	 * `hideIfTargetMenuEmpty` prop of `NavigationItem`.
+	 */
+	isEmpty?: boolean;
+	/**
+	 * Indicates whether the search is debouncing or not. In case of `true`, the
+	 * "No results found." text is omitted. Used to prevent showing the
+	 * "No results found." text between debounced searches.
+	 */
+	isSearchDebouncing?: boolean;
+	/**
+	 * The unique identifier of the menu.
+	 * The root menu can omit this prop, and it will default to "root".
+	 * All other menus need to specify it.
+	 *
+	 * @default 'root'
+	 */
+	menu?: string;
+	/**
+	 * The parent menu slug; used by nested menus to indicate their parent menu.
+	 */
+	parentMenu?: string;
+	/**
+	 * The menu title. It's also the field used by the menu search function.
+	 */
+	title?: string;
+	/**
+	 * Use this prop to render additional actions in the menu title.
+	 */
+	titleAction?: React.ReactNode;
+};
+
+export type NavigationGroupProps = {
+	/**
+	 * Optional classname for the component.
+	 */
+	className?: string;
+	/**
+	 * The group title.
+	 */
+	title?: string;
+	/**
+	 * The childred components.
+	 */
+	children: React.ReactNode;
+};
+
+type _NavigationItemButtonProps =
+	| ( ButtonProps & {
+			/**
+			 * If set to `true` the menu item will only act as a text-only item,
+			 * rather than a `<button />` or `<a />` element.
+			 */
+			isText?: false;
+	  } )
+	| {
+			/**
+			 * If set to `true` the menu item will only act as a text-only item,
+			 * rather than a `<button />` or `<a />` element.
+			 */
+			isText: true;
+	  };
+
+export type NavigationItemBaseProps = {
+	/**
+	 * The item badge content.
+	 */
+	badge?: string | number;
+	/**
+	 * The childred components. When not specified, the item will render the
+	 * default item UI.
+	 */
+	children?: React.ReactNode;
+	/**
+	 * Optional classname for the component.
+	 */
+	className?: string;
+	/**
+	 * If provided, causes the component to render an `<a />` element
+	 * instead of a `<button />` element.
+	 */
+	href?: string;
+	/**
+	 * If no `children` are passed, this prop allows to specify a custom icon for
+	 * the menu item.
+	 */
+	icon?: JSX.Element;
+	/**
+	 * The unique identifier of the item.
+	 */
+	item?: string;
+	/**
+	 * The child menu slug. If provided, clicking on the item will navigate
+	 * to the target menu.
+	 */
+	navigateToMenu?: string;
+	/**
+	 * Indicates whether this item should be hidden if the menu specified in
+	 * `navigateToMenu` is marked as empty in the `isEmpty` prop.
+	 * Used together with the `isEmpty` prop of `NavigationMenu`.
+	 */
+	hideIfTargetMenuEmpty?: boolean;
+	/**
+	 * A callback to handle clicking on a menu item.
+	 *
+	 * @default noop
+	 */
+	onClick?: React.MouseEventHandler;
+	/**
+	 * The item title.
+	 */
+	title?: string;
+};
+
+export type NavigationItemProps = _NavigationItemButtonProps &
```

Comment:
I would have liked to improve those types by adding the notion that the `children` and `icon` props shouldn't be both defined at the same time (since `icon` would be otherwise ignored), but I couldn't get to a working solution and therefore I (for now) gave up, for the sake of not delaying the TypeScript migration further.

Reply:
I'll leave that to your judgment. Definitely not a blocker, and likely not worth the effort if this component is going to be deprecated soon.